src/vmcu/services/communications/spi.py

```diff
@@ -18,41 +21,84 @@ class SPIPeripheral(ABC):
     To the other side, if you are a slave, you must check the command before processing the data.
     Communication itself has been implemented using UDP protocol.
     Messages are handled using queues and threads, one for transmission and other for reception.
-    '''
+    """
+
     @abstractmethod
-    def _send(self): pass
-    
+    def _send(self):
+        pass
+
     @abstractmethod
-    def _recv(self): pass
+    def _recv(self):
+        pass
 
-    def __init__(self, ip, port): 
-        self.socket = socket.socket(type=socket.SOCK_DGRAM) # Socket used to send and receive SPI data
+    def __init__(self, ip, port):
+        self.socket = socket.socket(
+            type=socket.SOCK_DGRAM
+        )  # Socket used to send and receive SPI data
         self.socket.bind((ip, port))
 
-        self.transmission_queue = queue.Queue() # Queue used to store messages that should be sended
-        self.reception_queue = queue.Queue() # Queue used to store received messages
-        self.transmission_thread = Thread(target=self._send , args=("Transmitter")) # Thread that manages data transmission
-        self.reception_thread = Thread(target=self._recv , args=("Receiver")) # Thread that manages data reception
+        self.transmission_queue = (
+            queue.Queue()
+        )  # Queue used to store messages that should be sended
+        self.reception_queue = queue.Queue()  # Queue used to store received messages
+        self.transmission_thread = Thread(
+            target=self._send, args=("Transmitter")
+        )  # Thread that manages data transmission
+        self.reception_thread = Thread(
+            target=self._recv, args=("Receiver")
+        )  # Thread that manages data reception
 
         self.transmission_thread.start()
         self.reception_thread.start()
 
-    def receive(self) -> bytes: 
+    def receive(self) -> bytes:
         return self.reception_queue.get
-    
+
     @abstractmethod
-    def transmit(self): pass
+    def transmit(self):
+        pass
 
     def __del__(self):
         self.stop = True
         self.transmission_thread.join()
         self.reception_thread.join()
         self.socket.close()
-        
 
-class SPIMaster(SPIPeripheral): 
+    def sendInput(self, data: bytes) -> Input:
+        return SPIInput(self, data)
+    
+    def sendPacket(self, packet: SPIPacket) -> Input:
+        return SPIInput(self, packet.data)
```

Comment:
I like the convention we have where the functions that return Inputs start with `generate_`, also this way if the logic changes for sending the data the rest of the functions stay the same
```suggestion
    def generate_send_raw(self, data: bytes) -> Input:
        return SPIInput(self, data)
    
    def generate_send_packet(self, packet: SPIPacket) -> Input:
        return generate_send_raw(self, packet.data)
```